### PR TITLE
feat: add support for mlflow experiment tracker

### LINKED ISSUE #315: feat: add support for mlflow experiment tracker
We have a very hard time getting an overview of:
* Which datasets have been generated, and how do we get to them?
* Which models have been trained, and how did they perform?

A simple MVP here is setting up MLFlow locally. Ideally, BI would host it, and it would be accessible via the VPN as an RM subdomain.

Awaiting #449 for Snyk scan, then we can reopen and merge #432.

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Fixes #315.

@HLasse Hold off on review, a new security vuln has been identified since I last looked at it.

### CODE REVIEW

File: psycop/common/model_training_v2/loggers/base_logger.py [hunk 1]
```diff
@@ -56,3 +59,57 @@ def log_config(self, config: dict[str, Any]) -> None:
...
+
+        # Write log temporarily to disk
+        with tmp_log_path.open("w") as f:
+            f.write(self._log_str)
+
+        # Overwrites any previously saved artifacts
+        mlflow.log_artifact(local_path=tmp_log_path.__str__(), artifact_path="log.txt")
+
+        # Delete the log on disk
+        tmp_log_path.unlink()
```
Comment (MartinBernstorff) [inline psycop/common/model_training_v2/loggers/base_logger.py:95]: A bit hacky here, @KennethEnevoldsen, do we know anyone with experience in mlflow?

Comment (KennethEnevoldsen) [reply psycop/common/model_training_v2/loggers/base_logger.py:95]: I sadly don't know anyone

File: psycop/common/model_training_v2/config/baseline_registry.py [hunk 2]
```diff
@@ -42,6 +38,3 @@ def to_dict(self) -> dict[str, catalogue.Registry]:
             for attribute_name in dir(self)
             if isinstance(getattr(self, attribute_name), catalogue.Registry)
         }
-
-
-populate_baseline_registry()
```
Comment (MartinBernstorff) [inline psycop/common/model_training_v2/config/baseline_registry.py:47]: Had to remove this to avoid circular imports with the TerminalLogger.

**Comment by MartinBernstorff** [comment]
A security vulnerability was identified in MLFlow, hindering this merge.

Report: https://huntr.com/bounties/7cf918b5-43f4-48c0-a371-4d963ce69b30/
On mlflow: https://github.com/mlflow/mlflow/issues/10533

**Comment by MartinBernstorff** [comment]
2.9 is being released this week. If it succeeds, we're ready to merge: https://github.com/mlflow/mlflow/issues/10533

**Comment by MartinBernstorff** [comment]
En yderligere: https://security.snyk.io/vuln/SNYK-PYTHON-MLFLOW-6069159
Detaljeret forklaring her: https://huntr.com/bounties/3e64df69-ddc2-463e-9809-d07c24dc1de4/

HVIS man bruger MLFLow som server eller UI (det gør vi ikke, det er BI der står for det)
OG en bruger på samme netværk (Region Midt) forsøger at oprette en bruger via en HTTP request
SÅ kan de tilgå filer der ligger på MLFlow serveren

Fordi det er BI der hoster MLFlow serveren er det ikke en sikkerhedssvaghed for os, da alle med adgang til MLFlow serveren også har adgang til vores server og data. Den potentielle svaghed er større, så vil gerne snakke om den :-)

**Comment by MartinBernstorff** [comment]
Afventer svar på undtagelse fra Andreas.

**Pull request merged**